salt/_modules/metalk8s_kubernetes.py

```diff
@@ -893,26 +893,48 @@ def delete_service(name, namespace='default', **kwargs):
 
     CLI Examples::
 
-        salt '*' kubernetes.delete_service my-nginx default
-        salt '*' kubernetes.delete_service name=my-nginx namespace=default
+        salt '*' kubernetes.delete_service my-nginx-svc
+        salt '*' kubernetes.delete_service name=my-nginx-svc namespace=default
     '''
     cfg = _setup_conn(**kwargs)
+    body = kubernetes.client.V1DeleteOptions()
 
     try:
-        api_instance = kubernetes.client.CoreV1Api()
-        body = kubernetes.client.V1DeleteOptions()
+        api_instance = kubernetes.client.kubernetes.client.CoreV1Api()
         api_response = api_instance.delete_namespaced_service(
             name=name,
-            body=body,
-            namespace=namespace)
-
-        return api_response.to_dict()
+            namespace=namespace,
+            body=body)
+        mutable_api_response = api_response.to_dict()
+        if not salt.utils.platform.is_windows():
+            try:
+                with _time_limit(POLLING_TIME_LIMIT):
+                    while show_service(name, namespace, **kwargs) is not None:
+                        time.sleep(1)
+                    else:  # pylint: disable=useless-else-on-loop
```

Comment:
Is is really an `useless-else-on-loop`.
If so, can we fix it instead of disabling the warning?

Reply:
it is not useless, there is a time limit `with _time_limit(POLLING_TIME_LIMIT)` and the while loop can finish "looping" without having the object back

Reply:
So remove the `# pylint: disable=useless-else-on-loop` then